src/main/java/org/jabref/gui/entryeditor/SourceTab.java

```diff
@@ -77,6 +78,15 @@ private Node createSourceEditor(BibEntry entry, BibDatabaseMode mode) {
             LOGGER.debug("Incorrect entry", ex);
         }
 
+        codeArea.richChanges().subscribe(x -> {
```

Comment:
I don't think it is a good idea if the sharelatex stuff depends on the entry editor. We have a text comparison engine that is used in the merge dialog and directly gives you the information you need (what characters changed at which position).